explorer/explorer.go

```diff
@@ -308,23 +308,23 @@ func (exp *explorerUI) Store(blockData *blockdata.BlockData, val *wire.MsgBlock)
 		go exp.prePopulateChartsData()
 	}
 
-	// Lock for explorerUI's NewBlockData and ExtraInfo
-	exp.NewBlockDataMtx.Lock()
-
 	// Returns the Block with some more data needed for the full block visualization.
-	exp.NewBlockData = exp.blockData.GetExplorerBlock(val.BlockHash().String())
-	bdHeight := exp.NewBlockData.Height
+	newBlockData := exp.blockData.GetExplorerBlock(msgBlock.BlockHash().String())
+	targetTimePerBlock := float64(exp.ChainParams.TargetTimePerBlock)
 	difficulty := blockData.Header.Difficulty
+	bdHeight := newBlockData.Height
 
+	// 248 blocks take almost a day to mine
+	last24hrDifficulty := exp.blockData.DifficultyAtHeight(bdHeight - 248)
```

Comment:
Typo here, I think.  It's 288 blocks/day.  But we want to compute this from `exp.ChainParams.TargetTimePerBlock`

Reply:
or I could use the actual timestamp from 24hrs ago to fetch the difficulty value.

Reply:
Honestly I don't know what's right.  I would check in #dev and/or with @davecgh